test/TestProjects/ModelShapes/Generated/SchemaMappingModelFactory.cs

```diff
@@ -0,0 +1,84 @@
+        public static MixedModel MixedModel(string requiredString, int requiredInt, IList<string> requiredStringList, IList<int> requiredIntList, string nonRequiredString, int? nonRequiredInt, IList<string> nonRequiredStringList, IList<int> nonRequiredIntList, string requiredNullableString, int? requiredNullableInt, IList<string> requiredNullableStringList, IList<int> requiredNullableIntList, string nonRequiredNullableString, int? nonRequiredNullableInt, IList<string> nonRequiredNullableStringList, IList<int> nonRequiredNullableIntList, int requiredReadonlyInt, int? nonRequiredReadonlyInt)
+        {
+            return new MixedModel(requiredString, requiredInt, requiredStringList, requiredIntList, nonRequiredString, nonRequiredInt, nonRequiredStringList, nonRequiredIntList, requiredNullableString, requiredNullableInt, requiredNullableStringList, requiredNullableIntList, nonRequiredNullableString, nonRequiredNullableInt, nonRequiredNullableStringList, nonRequiredNullableIntList, requiredReadonlyInt, nonRequiredReadonlyInt);
+        }
+
+        /// <summary> Initializes new instance of OutputModel class. </summary>
+        /// <param name="requiredString"> . </param>
+        /// <param name="requiredInt"> . </param>
+        /// <param name="requiredStringList"> . </param>
+        /// <param name="requiredIntList"> . </param>
+        /// <param name="nonRequiredString"> . </param>
+        /// <param name="nonRequiredInt"> . </param>
+        /// <param name="nonRequiredStringList"> . </param>
+        /// <param name="nonRequiredIntList"> . </param>
+        /// <param name="requiredNullableString"> . </param>
+        /// <param name="requiredNullableInt"> . </param>
+        /// <param name="requiredNullableStringList"> . </param>
+        /// <param name="requiredNullableIntList"> . </param>
+        /// <param name="nonRequiredNullableString"> . </param>
+        /// <param name="nonRequiredNullableInt"> . </param>
+        /// <param name="nonRequiredNullableStringList"> . </param>
+        /// <param name="nonRequiredNullableIntList"> . </param>
+        /// <param name="requiredReadonlyInt"> . </param>
+        /// <param name="nonRequiredReadonlyInt"> . </param>
+        /// <returns> A new <see cref="Models.OutputModel"/> instance for mocking. </returns>
+        public static OutputModel OutputModel(string requiredString, int requiredInt, IReadOnlyList<string> requiredStringList, IReadOnlyList<int> requiredIntList, string nonRequiredString, int? nonRequiredInt, IReadOnlyList<string> nonRequiredStringList, IReadOnlyList<int> nonRequiredIntList, string requiredNullableString, int? requiredNullableInt, IReadOnlyList<string> requiredNullableStringList, IReadOnlyList<int> requiredNullableIntList, string nonRequiredNullableString, int? nonRequiredNullableInt, IReadOnlyList<string> nonRequiredNullableStringList, IReadOnlyList<int> nonRequiredNullableIntList, int requiredReadonlyInt, int? nonRequiredReadonlyInt)
```

Comment:
We should have default values for non-required properties but also maybe for all of them.

Reply:
What should be default for lists? `null` or empty lists?

Reply:
I think empty lists. Similar to how when you create models lists are already initialized.

Reply:
So the logic is this:
- If parameter has default value declared in constructor, use that default value
- If parameter doesn't have declared default value, use empty list/dictionary for list/dictionary types and `default` otherwise

Reply:
Fixed